### PR TITLE
feat(Text input group): added autocomplete ghosting

### LINKED ISSUE #4558: Add autocomplete ghosting feature to TextInputGroup
similar to what is done in SearchInput

### PR DESCRIPTION
**Pull request by evwilkin**
Closes #4558 

This PR adds a new example for text input group using the existing `--pf-c-text-input-group__text-input--m-hint--Color` variable to show ghosting for autocomplete suggestion.

### CODE REVIEW

**Review by mattnolting — APPROVED** [review]
L🍕 TM

File: src/patternfly/components/TextInputGroup/text-input-group-text-input.hbs [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
   type="text"
-  value="{{#if text-input-group--IsDisabled}}Disabled{{else}}{{text-input-group--value}}{{/if}}"
+  {{#if text-input-group--IsDisabled}}
+    value="Disabled"
+  {{else if text-input-group-text-input--hint-text}}
+    value="{{text-input-group-text-input--hint-text}}"
+  {{else}}
+    value="{{text-input-group--value}}"
+  {{/if}}
   aria-label="Type to filter"
```
Comment (mcoker) [inline src/patternfly/components/TextInputGroup/text-input-group-text-input.hbs:14]: Can we omit this if it's a hint input? It won't do anything if it's `aria-hidden`. Looks like the react component is already doing that in the search input. We'll need to update the search input, too - we can do that here or in a separate issue if you'd prefer.

```suggestion
  {{#if text-input-group-text-input--hint-text }}
    aria-label="Type to filter"
  {{/unless}}
```

**Review by mcoker — APPROVED** [review]
Fabulous! A couple of nits, neither are blockers if we want to follow up later.

**Comment by evwilkin** [comment]
@mcoker made your suggested changes - will need a separate follow up for SearchInput, I took a look but wasn't quite sure what needed to change.

**Pull request merged**